code/framework/object/src/main/java/io/cattle/platform/object/meta/impl/DefaultObjectMetaDataManager.java

```diff
@@ -683,6 +683,25 @@ public Relationship getRelationship(String type, String linkName) {
     }
 
     @Override
+    public Relationship getRelationship(String type, String linkName, String fieldName) {
+        Class<?> clz = schemaFactory.getSchemaClass(type, true);
+        Map<String, List<Relationship>> relationship = relationshipsBothCase.get(clz);
+        List<Relationship> relationshipList = relationship.get(linkName);
+        if (relationshipList == null || relationshipList.size() == 0) {
+            return null;
+        }
+        if (relationshipList.size() > 1) {
+            for (Relationship rel : relationshipList) {
+                if (rel instanceof ForeignKeyRelationship && rel.getName() != null && rel.getName().equalsIgnoreCase(linkName) && rel.getPropertyName().equalsIgnoreCase(fieldName)) {
+                    return rel;
+                }
+            }
+            return null;
+        }
```

Comment:
Not sure, if my change needs to be something else as this is an edge case that seemed to be specific enough, eg having a foreign key relationship to the same table with two different fields on a link and needing to be able to chose the fieldName.